nikola/plugin_categories.py

```diff
@@ -304,7 +304,10 @@ def compile(self, source, dest, is_two_file=True, post=None, lang=None):
         self.compile_html(source, dest, is_two_file)
 
     def compile_string(self, data, source_path=None, is_two_file=True, post=None, lang=None):
-        """Compile the source file into HTML strings (with shortcode support)."""
+        """Compile the source file into HTML strings (with shortcode support).
+
+        Returns a tuple of at least two elements (html string, shortcode dependencies).
```

Comment:
Someone might interpret this that shortcode dependencies should always be the second element in the list (which is not true). Should we change the implementations to always return this as second, or adjust the documentation?